xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -141,32 +168,251 @@ bool CDVDSubtitlesLibass::DecodeDemuxPkt(const char* data, int size, double star
+    }
+
+    ass_set_selective_style_override_enabled(m_renderer, stylesFlags);
+    ass_set_selective_style_override(m_renderer, style);
+  }
+}
+
+ASS_Style* CDVDSubtitlesLibass::CreateStyleDefault(double scale)
+{
+  const std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+
+  // Allocate and define a new style used on each event
+  int n = ass_alloc_style(m_track);
+  m_track->default_style = n;
+  ASS_Style* style = m_track->styles + n;
+  style->Name = strdup("KodiDefault");
+
+  style->FontName = strdup(DEFAULT_FONT_NAME); // Font name is case sensitive
+  RESOLUTION_INFO res = CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo(
+      CServiceBroker::GetWinSystem()->GetGfxContext().GetVideoResolution());
+  int settingFontSize = settings->GetInt(CSettings::SETTING_SUBTITLES_HEIGHT);
+  style->FontSize = static_cast<double>(settingFontSize) * scale;
+  // Modifies the width/height of the font (in %)
+  style->ScaleX = 1.;
+  style->ScaleY = 1.;
+  // Extra space between characters causes the underlined
+  // text line to become more discontinuous (test on LibAss 15.1)
+  style->Spacing = 0;
+
+  uint32_t fontStyleMask = settings->GetInt(CSettings::SETTING_SUBTITLES_STYLE) & FONT_STYLE_MASK;
+  if (fontStyleMask & FONT_STYLE_BOLD)
+    style->Bold = 1;
+  if (fontStyleMask & FONT_STYLE_ITALICS)
+    style->Italic = 1;
+
+  // Compute the color to be used for the font color (depending on the opacity)
+  UTILS::Color subColor = colors[settings->GetInt(CSettings::SETTING_SUBTITLES_COLOR)];
+  int subOpacity = settings->GetInt(SETTING_SUBTITLES_OPACITY);
+  subColor = ColorUtils::ChangeOpacity(subColor, (100 - subOpacity) / 100.0f);
+
+  // Default subtitles color
+  style->PrimaryColour = ColorUtils::ConvertToRGBA(subColor);
+  // SecondaryColour may be used to prevent an onscreen collision
+  style->SecondaryColour = ColorUtils::ConvertToRGBA(subColor);
+  // Outline color
+  UTILS::Color outlineColor =
+      ColorUtils::ConvertToRGBA(ColorUtils::ChangeOpacity(UTILS::COLOR::BLACK, 0));
+
+  // NOTE: All colors properties must be set always just after the border style property
+  int bgOpacity = settings->GetInt(CSettings::SETTING_SUBTITLES_BGOPACITY);
+  if (bgOpacity == 0) // If no background, enable the outlines
+  {
+    // The outlines currently can be enabled only with BorderStyle=1
+    style->BorderStyle = 1; // Outline + drop shadow
+    style->OutlineColour = outlineColor;
+    style->BackColour = ColorUtils::ConvertToRGBA(
+        ColorUtils::ChangeOpacity(UTILS::COLOR::BLACK, 0)); // Shadow color
+    style->Outline = 1 * scale;
+    style->Shadow = 0;
+  }
+  else
+  {
+    // Compute the color to be used for the background (depending on the opacity)
+    UTILS::Color bgColor = bgColors[settings->GetInt(CSettings::SETTING_SUBTITLES_BGCOLOR)];
+    bgColor = ColorUtils::ChangeOpacity(bgColor, (100 - bgOpacity) / 100.0f);
+
+    style->BorderStyle = 4; // Opaque box + outline
```

Comment:
Please avoid magic numbers, define const values for them

Reply:
done